libsignal-service/src/envelope.rs

```diff
@@ -154,23 +151,16 @@ impl Envelope {
     }
 
     pub fn source_address(&self) -> ServiceAddress {
-        let phonenumber = self
-            .source_e164
-            .as_ref()
-            .map(|s| phonenumber::parse(None, s))
-            .transpose()
-            .expect("valid e164 checked in constructor");
-
         let uuid = self
             .source_uuid
             .as_deref()
             .map(Uuid::parse_str)
             .transpose()
             .expect("valid e164 checked in constructor");
+
         ServiceAddress {
-            phonenumber,
+            phonenumber: None,
```

Comment:
Even more scary. Should we modify `ServiceAddress` to be a wrapper around `Uuid`?

Reply:
It's still in use in other parts of the code IIRC, so not all code paths are e164-less yet. I'll double check. (I think it was unidentified message somewhere).

Reply:
Let's make it an issue that we remove the phone number field, if that's the same in Android/iOS.